elastic/eui
# [Emotion] Convert EuiButtonIcon

## Summary

This PR (hopefully) should be fairly quick to review/verify. Not a lot of major changes going on here (other than a million snapshots).

- Removes internal `useEuiButtonRadiusCSS` hook in favor of a more agnostic `euiButtonSizeMap` - using the raw size data is less strict & generates fewer classNames
- Converts `EuiButtonIcon` to Emotion, removing/cleaning up unnecessary CSS
- Removes `.euiButtonIcon--[size]` modifier classes
- Fixes `isLoading` button icon spinners to have the same background color contrast as other button loading spinners

## QA

- Compare [production](https://elastic.github.io/eui/#/navigation/button#icon-buttons) against [staging](https://eui.elastic.co/pr_6844/#/navigation/button#icon-buttons) and confirm that `EuiButtonIcon`'s demo example looks the same between the two
- `gh pr checkout 6844 && yarn storybook`
- Go to the `EuiButtonIcon` playground and test multiple color/size permutations and confirm they look the same as before

### General checklist


## Emotion checklist

### Acceptance criteria
~- [ ] All SCSS overrides have been removed from the Amsterdam theme~ - No overrides for this component
~- [ ] Any dependent components are identified in a new issue~ - All dependent usages in EUI are calling the generic `.euiButtonIcon`

### Checklists

**Kibana usage**

  ~- [ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~ - no major Kibana usages

---

**General**


---

**Unit tests**

- [x] Converted from Enzyme to RTL
~- [ ] Removed any mounted snapshots~

---

**Sass/Emotion conversion process**


~- [ ] Deleted any `src/amsterdam/overrides/{component}.scss` files (styles within should have been converted to the baseline Emotion styles)~
~- [ ] Converted all global Sass vars/mixins to JS (e.g. `$euiSize` to `euiTheme.size.base`)~
~- [ ] Removed or converted component-specific Sass vars/mixins to exported JS versions~
~- [ ] Listed var/mixin removals in changelog~
~- [ ] Ran `yarn compile-scss` to update var/mixin [JSON files](https://github.com/elastic/eui/tree/main/src-docs/src/views/theme/_json)~
~- [ ] Simplified `calc()` to `mathWithUnits` if possible (if mixing different unit types, this may not be possible)~
~- [ ] Added an `@warn` deprecation message within the `global_styling/mixins/{component}.scss` file~

---

**CSS tech debt**

~- [ ] Used `gap` property to add margin between items if using flex~
~- [ ] Wrapped all animations or transitions in `euiCanAnimate`~

---

**DOM Cleanup**


---

**Kibana due diligence**

~- [ ] Any Sass or CSS that will need to be updated, particularly if a component Sass var was deleted~ None
    - 2 minor usages as test queries that will need to be updated
    - [1 usage that will need to be fixed](https://github.com/elastic/kibana/blob/ebac6946122355ab25b226642e5aa9dab7c7710b/x-pack/plugins/maps/public/connected_components/right_side_controls/layer_control/layer_control.tsx#L52)

---

**Extras/nice-to-have**

~- [ ] Documentation pass~
~- [ ] Check for issues in the backlog that could be a quick fix for that component~
~- [ ] Optional component/code cleanup: consider splitting up the component into multiple children if it's overly verbose or difficult to reason about~
